build-common/src/org/jetbrains/kotlin/incremental/storage/externalizers.kt

```diff
@@ -27,40 +27,33 @@ import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.resolve.jvm.JvmClassName
 import java.io.*
 
-/**
- * Storage versioning:
- * 0 - only name and value hashes are saved
- * 1 - name and scope are saved
- */
 class LookupSymbolKeyDescriptor(
     /** If `true`, original values are saved; if `false`, only hashes are saved. */
     private val storeFullFqNames: Boolean = false
 ) : KeyDescriptor<LookupSymbolKey> {
 
     override fun read(input: DataInput): LookupSymbolKey {
-        val version = input.readByte()
-        return when (version.toInt()) {
-            0 -> {
-                val name = input.readUTF()
-                val scope = input.readUTF()
-                LookupSymbolKey(name.hashCode(), scope.hashCode(), name, scope)
-            }
-            1 -> {
-                val first = input.readInt()
-                val second = input.readInt()
-                LookupSymbolKey(first, second, "", "")
-            }
-            else -> throw IllegalArgumentException("Unknown version of LookupSymbolKeyDescriptor=${version}")
+        // Note: The value of the storeFullFqNames variable below may or may not be the same as LookupSymbolKeyDescriptor.storeFullFqNames.
+        // 0 means storeFullFqNames == true, see `save` function below.
+        val storeFullFqNames = (input.readByte() == 0.toByte())
```

Comment:
Are only 0 and 1 allowed values? In such case it'd be nice to check for that condition.

Reply:
Yeah I added a check.